src/ucp/wireup/wireup_cm.c

```diff
@@ -1176,47 +1193,20 @@ ucp_ep_server_init_priv_data(ucp_ep_h ep,  const char *dev_name,
                                  goto out;
                              });
 
-    tl_bitmap = ucp_ep_get_tl_bitmap(ep);
+    ucp_ep_get_tl_bitmap(ep, &tl_bitmap);
+    ucp_context_dev_tl_bitmap(worker->context, dev_name, &ctx_tl_bitmap);
 
     /* make sure that all lanes are created on correct device */
-    ucs_assert(UCS_BITMAP_IS_ZERO(
-            UCP_TL_BITMAP_AND_NOT(
-                    tl_bitmap, ucp_context_dev_tl_bitmap(worker->context,
-                                                         dev_name)),
-            UCP_MAX_RESOURCES));
+    ucs_assert(UCS_BITMAP_IS_ZERO(UCP_TL_BITMAP_AND_NOT(tl_bitmap,
+                                                        ctx_tl_bitmap),
+                                  UCP_MAX_RESOURCES));
 
-    status = ucp_address_pack(worker, ep, &tl_bitmap,
-                              ucp_cm_address_pack_flags(worker), NULL,
-                              &ucp_addr_size, &ucp_addr);
-    if (status != UCS_OK) {
-        goto out;
-    }
-
-    if (worker->cms[ucp_ep_ext_control(ep)->cm_idx].attr.max_conn_priv <
-        ucp_cm_priv_data_length(ucp_addr_size)) {
-        status = UCS_ERR_BUFFER_TOO_SMALL;
-        goto free_addr;
-    }
+    dev_index = ucp_cm_tl_bitmap_get_dev_idx(worker->context, &tl_bitmap);
+    status    = ucp_cm_ep_priv_data_pack(ep, &tl_bitmap, dev_index, data_buf_p,
```

Comment:
how can we sure ucp_cm_ep_priv_data_pack will not try to do fallback on server side?

Reply:
AFAIK fallback initiated by server side is not supported now, it would mean "rejected" for client or active/passive role change

Reply:
right, but seems there is a flow ucp_cm_ep_priv_data_pack() may initiate a fallback?

Reply:
only client side, server side ucp_ep_create will just return error